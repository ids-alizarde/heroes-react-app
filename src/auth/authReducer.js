import { CONSTANTS } from "../services/constantsService";

// const state = {
//     name: 'Aldo',
//     logged: true
// };

export const authReducer = ( state = {}, action ) => {

    switch ( action.type ) {
        case CONSTANTS.login:
            
            return {
                ...action.payload,
                logged: true
            };

        case CONSTANTS.logout:
            
            return {
                logged: false
            };

        default:
            return state;
    }
}