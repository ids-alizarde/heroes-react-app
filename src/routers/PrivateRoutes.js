import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';

export const PrivateRoutes = ({
    isAuthenticated,
    component: Component,
    ...rest
}) => {

    let lastPathVisited = rest.location.pathname;
    const queryString = rest.location.search;

    lastPathVisited = ( queryString.length > 3 ) ? lastPathVisited + queryString : lastPathVisited;

    localStorage.setItem( 'lastPath', lastPathVisited );
    
    return (
        <Route { ...rest }
            component={( props ) => (
                ( isAuthenticated ) 
                    ? <Component { ...props } />
                    : <Redirect to="/login" />
            )}
        />
    )
}

PrivateRoutes.propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    component: PropTypes.func.isRequired
}
