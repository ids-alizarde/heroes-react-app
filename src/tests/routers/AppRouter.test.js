import React from 'react';
import { mount } from 'enzyme';
import { AuthContext } from '../../auth/AuthContext';
import { AppRouter } from '../../routers/AppRouter';


describe('Pruebas en <AppRouter />', () => {
    
    test('Debe de mostrar el componente LoginScren si no esta autenticado', () => {

        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: false
            }
        };
        
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'button' ).exists() ).toBe( true );
    });

    test('Debe de mostrar el componente MarvelScreen si esta autenticado', () => {

        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Matias'
            }
        };
        
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect( wrapper.find( '.navbar' ).exists() ).toBe( true );
    });
    
})
