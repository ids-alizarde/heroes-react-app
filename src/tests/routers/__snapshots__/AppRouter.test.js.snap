// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> Debe de mostrar el componente LoginScren si no esta autenticado 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login Screen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Ingresar
    </button>
  </div>
</div>
`;
